kernel-modules/probe/.clang-format

```diff
@@ -0,0 +1,25 @@
+BasedOnStyle: Google
```

Comment:
I see this file is based on `collector/.clang-format`, would it make sense to use a symlink to that file so we don't need to maintain 2 versions of the same file? I know that might be troublesome for docker builds, but the tradeoff might be worth it.

Reply:
Might have to check actually how clang-format works because I have a feeling if we put it in the repo root directory then it'll affect all files recursively (except those with another .clang-format file in their local directory). If that is the case I'll move it to the repo root so we don't have to worry about symlinks

Reply:
Looks like that doesn't work, unfortunately. A symlink might be our only option, so my thinking now is to have the file at the root of the repository and then symlink that into `kernel-modules/probe` and `collector`. What do you think?

Reply:
Apparently I spoke too soon - putting the file in the root directory does appear to work